File: app/src/main/java/com/itech/wisherbirthdaywishapp/views/ui/friendprofilefragment/GiftRepository.java
package com.itech.wisherbirthdaywishapp.views.ui.friendprofilefragment;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.itech.wisherbirthdaywishapp.database.GiftDatabase;
import com.itech.wisherbirthdaywishapp.model.dao.GiftDao;
import com.itech.wisherbirthdaywishapp.model.entities.Gift;
import com.itech.wisherbirthdaywishapp.model.entities.relations.GiftAndWish;

import java.util.List;

public class GiftRepository {
    private GiftDao giftDao;
    private LiveData<List<Gift>> mAllGifts;
    private LiveData<List<GiftAndWish>> mAllGiftsWithWishes;

    public GiftRepository(Application application) {
        GiftDatabase giftDatabase = GiftDatabase.Companion.getInstance(application);
        giftDao = giftDatabase.getGiftDao();
        mAllGiftsWithWishes = giftDao.getGiftsWithWishes();
        mAllGifts = giftDao.getGifts();
    }

    public LiveData<List<Gift>> getAllGifts() {
        return mAllGifts;
    }

    public LiveData<List<GiftAndWish>> getAllGiftsWithWishes() {
        return mAllGiftsWithWishes;
    }

    public void insert(Gift gift) {
        GiftDatabase.Companion.getExecutor().execute(() ->{
            giftDao.insertGift(gift);
        });
    }
}

File: app/src/main/java/com/itech/wisherbirthdaywishapp/repositories/GiftRepository.kt
package com.itech.wisherbirthdaywishapp.repositories

import android.content.Context
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Transaction
import com.itech.wisherbirthdaywishapp.database.GiftDatabase
import com.itech.wisherbirthdaywishapp.model.dao.GiftDao
import com.itech.wisherbirthdaywishapp.model.entities.Gift
import com.itech.wisherbirthdaywishapp.model.entities.Wish
import com.itech.wisherbirthdaywishapp.model.entities.relations.GiftAndWish

class GiftRepository( val context: Context) {
    private val giftDao: GiftDao = GiftDatabase.getInstance(context).giftDao

    suspend fun insertWish(wish: Wish) {
        giftDao.insertWish(wish)
    }

    suspend fun insertGift(gift: Gift) {
        giftDao.insertGift(gift)
    }

    suspend fun getGiftsWithoutWishes(): List<Gift> {
        return giftDao.getGiftsWithoutWishes()
    }

    suspend fun getGiftsWithWishes(): List<GiftAndWish> {
        return giftDao.getGiftsWithWishes()
    }

    suspend fun getGiftWithWishById(id: String): GiftAndWish {
        return giftDao.getGiftWithWishById(id)
    }

}